File: TennisKata/JS/tennisCounter.js
playerOneScore = 0;
playerTwoScore = 0;

function Score()
{
	var dict = ["Love","Fifteen","Thirty","Forty"];
	if(playerOneScore == playerTwoScore)
	{
		if(playerOneScore>=3) return "Deuce";
		return dict[playerOneScore] + " All";
	}
	else
	{
		if(playerOneScore < 4 && playerTwoScore < 4)
		{
			return dict[playerOneScore] + " " + dict[playerTwoScore];
		}
		else
		{
			if(Math.abs(playerOneScore-playerTwoScore) < 2)
			{
				if(playerOneScore>playerTwoScore)
				{
					return "PlayerA Adv";
				}
				else
				{
					return "Player2 Adv";
				}
			}
			else
			{
				if(playerOneScore>playerTwoScore)
				{
					return "PlayerA Won";
				}
				else
				{
					return "Player2 Won";
				}

			}			
		}
	}	
}

function FirstPlayerScore()
{
	if(!isMatchEnd()) playerOneScore++;
	updateview();
}

function SecondPlayerScore()
{
	if(!isMatchEnd()) playerTwoScore++;
	updateview();
}

function isMatchEnd()
{
	if(playerOneScore >= 4 && playerOneScore - playerTwoScore >= 2) return true;
	if(playerTwoScore >= 4 && playerTwoScore - playerOneScore >= 2) return true;
	return false;
}

function ResetScore()
{
	playerOneScore=0;
	playerTwoScore=0;
	updateview();
}

function updateview()
{
	console.log(444);
	document.getElementById('playerOneScore').innerHTML = playerOneScore;
	document.getElementById('playerTwoScore').innerHTML = playerTwoScore;
	document.getElementById('matchResult').innerHTML = Score();
}

File: TennisKata/C#TDD/TennisKata_TDD/TennisKata_TDD/TennisCounter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TennisKata_TDD
{
    class TennisCounter
    {
        private int _firstPlayerScore;
        private int _secondPlayerScore;

        public string ScoreMsg()
        {
            Dictionary<int,string> ScoreString = new Dictionary<int,string>()
            {
                {0,"Love"},
                {1,"Fifteen"},
                {2,"Thirty"},
                {3,"Forty"},
            };
            if (_firstPlayerScore == _secondPlayerScore && _firstPlayerScore >= 3) return "Deuce";
            if (_firstPlayerScore == _secondPlayerScore) return ScoreString[_secondPlayerScore] + " All";
            if (_firstPlayerScore >= 4 && _firstPlayerScore > _secondPlayerScore )
            {
                if (_firstPlayerScore - _secondPlayerScore == 1)
                    return "Player1 Adv";
                else
                    return "Player1 Won";
            }

            if (_secondPlayerScore >= 4)
            {
                if (_secondPlayerScore - _firstPlayerScore >= 2)
                    return "Player2 Won";
                else
                    return "Player2 Adv";
            }

            return ScoreString[_firstPlayerScore] + " " + ScoreString[_secondPlayerScore];    
        }

        public void FirstPlayerScore(int score)
        {
            _firstPlayerScore += score;
        }
        public void SecondPlayerScore(int score)
        {
            _secondPlayerScore += score;
        }
    }
}
